web/app/scripts/utils/date-utils.js

```diff
@@ -0,0 +1,87 @@
+(function() {
+    'use strict';
+
+    /* ngInject */
+    function DateUtils(LanguageState) {
+        var languageMap = {
+            'ar-sa': {
+                language: 'ar',
+                calendar: 'ummalqura',
+                formats: {
+                    'short': 'M Y',
+                    'long': 'd MM, Y',
+                    'numeric': ' - dd/mm/yyyy'
+                }
+            },
+            'en-us': {
+                language: 'en',
+                calendar: 'gregorian',
+                formats: {
+                    'short': 'M Y',
+                    'long': 'MM d, Y',
+                    'numeric': 'mm/dd/yyyy - '
+                }
+            },
+            'exclaim': {
+                language: 'en',
+                calendar: 'gregorian',
+                formats: {
+                    'short': '\'!\'M Y',
+                    'long': '\'!\'MM d, Y',
+                    'numeric': '\'!\'mm/dd/yyyy - '
+                }
+            }
+        };
+
+        var module = {
+            getLocalizedDateString: getLocalizedDateString
+        };
+        return module;
+
+        function getLocalizedDateString(date, format, includeTime) {
+            var language = LanguageState.getSelected().id;
+            if (!language) {
+                language = 'exclaim';
+            }
+            var isRtl = LanguageState.getSelected().rtl;
+            var conversion = languageMap[language];
+            var parsedDate = $.calendars.instance()
+                .parseDate('mm/dd/yyyyy', date.toLocaleDateString()).toJSDate();
```

Comment:
I'm not sure this is going to work in all cases; the docs for `toLocaleDateString` specify that it returns a "language-sensitive" date representation, so if a user's browser is in a language other than `en-us` I would expect this to return a format other than `mm/dd/yyyy`, which would cause parsing to fail.

In any event, I think this line may be superfluous; since Javascript Date objects are already in the Gregorian calendar, `date` can be used directly in line 50:
`var convertedDate = $.calendars.instance(conversion.calendar, conversion.language).fromJSDate(date);`